src/ansys/mechanical/core/embedding/license_manager.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2022 - 2025 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""License Manager."""
+from ansys.mechanical.core import LOG
+
+# TODO: enable logging
+
+
+class LicenseManager:
+    """Message manager for adding, fetching, and printing messages."""
+
+    def __init__(self, app):
+        """Initialize the message manager."""
+        self._app = app
+        if self._app.version < 252:
+            LOG.warning("License manager is only available in Ansys 2025 R2 and later.")
+            return
+
+        self._license_preference = self._app.ExtAPI.Application.LicensePreference
+
+    def _get_all_licenses(self):
+        """Return all active licenses."""
+        return self._license_preference.GetAllLicenses()
+
+    def get_license_status(self, license_name: str):
+        """Return the status of a license."""
```

Comment:
If it's a bool, the following docstring is more helpful: "Return whether the license is active."